### PR TITLE
Add should.reverting.withMessage helper

### PR DESCRIPTION
**Pull request by remon-nashid**
Same PR content of the following PR, while adapting to web3 1.0.

https://github.com/OpenZeppelin/openzeppelin-solidity/pull/1540

### CODE REVIEW

**Comment by remon-nashid** [comment]
One integration test fails because `ypeError: BigNumber is not a constructor`. I will investigate this tomorrow.

**Comment by nventuro** [comment]
The integration tests are currently broken :( so don't worry too much about that failure

File: src/shouldFail.js [hunk 1]
```diff
@@ -1,4 +1,5 @@
 const { should } = require('./setup');
+const semver = require('semver');
```
Comment (nventuro) [inline src/shouldFail.js]: We should add this to the `package.json`.

File: src/shouldFail.js [hunk 2]
```diff
@@ -29,7 +30,22 @@ async function shouldFail (promise) {
   await shouldFailWithMessage(promise);
 }
 
+async function withMessage (promise, message) {
+  // Find out if current version of ganache-core supports revert reason i.e >= 2.2.0.
```
Comment (nventuro) [inline src/shouldFail.js]: Can we add a link to the 2.2.0 release notes? https://github.com/trufflesuite/ganache-core/releases/tag/v2.2.0

File: src/shouldFail.js [hunk 3]
```diff
@@ -29,7 +30,22 @@ async function shouldFail (promise) {
...
+  const web3Version = await web3.eth.getNodeInfo();
```
Comment (nventuro) [inline src/shouldFail.js]:
```suggestion
  const nodeInfo = await web3.eth.getNodeInfo();
```

File: src/shouldFail.js [hunk 4]
```diff
@@ -29,7 +30,22 @@ async function shouldFail (promise) {
...
+  const matches = /TestRPC\/v([0-9.]+)\/ethereum-js/.exec(web3Version);
```
Comment (nventuro) [inline src/shouldFail.js]:
```suggestion
  const matches = /TestRPC\/v([0-9.]+)\/ethereum-js/.exec(nodeInfo);
```

File: test/src/shouldFail.test.js [hunk 5]
```diff
@@ -26,6 +26,10 @@ describe('shouldFail', function () {
       await shouldFail(this.failer.failWithRevert());
     });
 
+    it('accepts a require() revert', async function () {
+      await shouldFail(this.failer.failRequirement());
```
Comment (nventuro) [inline test/src/shouldFail.test.js]: Can we also add this case to the `reverting` `describe`?

File: test/src/shouldFail.test.js [hunk 6]
```diff
@@ -51,6 +55,20 @@ describe('shouldFail', function () {
     it('rejects an outOfGas', async function () {
       await assertFailure(shouldFail.reverting(this.failer.failWithOutOfGas({ gas: 2000000 })));
     });
+
+    describe('reverting.withMessage', function () {
+      it('rejects if no failure occurs', async function () {
+        await assertFailure(shouldFail.reverting.withMessage(this.failer.dontFail()));
```
Comment (nventuro) [inline test/src/shouldFail.test.js]: Can we also add a test case where a revert _does_ occur, but the message is wrong?

**Comment by remon-nashid** [comment]
Love your reviews @nventuro! Please take another look and let me know how PR looks now.

**Comment by remon-nashid** [comment]
I updated readme.md with a usage example. I will follow up with an integration test PR as soon as I understand how they work.

File: README.md [hunk 7]
```diff
@@ -19,7 +19,7 @@ const { BN, constants, expectEvent, shouldFail } = require('openzeppelin-test-he
 const ERC20 = artifacts.require('ERC20');
 
 contract('ERC20', ([sender, receiver]) => {
-  beforeEaach(async function () {
+  beforeEach(async function () {
```
Comment (nventuro) [inline README.md]: :grin: thanks!

File: README.md [hunk 8]
```diff
@@ -124,6 +124,45 @@ Collection of assertions for failures (similar to [chai's `throw`](https://www.c
...
+
+  describe('doOwnerOperation', function() {
+    it('Fails when called by a non-owner account', async function () {
+      await shouldFail.reverting.withMessage(this.owned.doOwnerOperation({ from: other }), "Unauthorized");
+    });
+  });
+  ...
+```
+
+Use this helper to specify expected error message, when you're testing a function that can revert for multiple reasons.
```
Comment (nventuro) [inline README.md]:
```suggestion
Use this helper to specify the expected error message, when you're testing a function that can revert for multiple reasons.
```

Comment (remon-nashid) [reply README.md]: Done.

**Comment by nventuro** [comment]
Woohoo!

**Pull request merged**